openedx/features/pakx/lms/pakx_admin_app/views.py

```diff
@@ -0,0 +1,132 @@
+
+                    return Response(status=status.HTTP_200_OK)
+
+                else:
+                    for update_id in request.data["ids"]:
+                        if not User.objects.filter(id=update_id).update(is_active=activation_status):
+                            return Response({"ID not found": update_id}, status=status.HTTP_404_NOT_FOUND)
+
+                    return Response(status=status.HTTP_200_OK)
+            else:
+                return Response(status=status.HTTP_400_BAD_REQUEST)
+        except:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+    def destroy(self, request, *args, **kwargs):
+        if User.objects.filter(id=self.get_object().id).update(is_active=False):
+            return Response(status=status.HTTP_200_OK)
+        else:
+            return Response({"ID not found": self.get_object().id}, status=status.HTTP_404_NOT_FOUND)
+
+    def list(self, request, *args, **kwargs):
+        if len(self.request.query_params) == 0:
+            return Response(self.serializer_class(self.queryset.order_by('-id'), many=True).data)
+
+        else:
+            queryset = None
+
+            if self.request.query_params.get('roles', None) is not None:
+                roles = self.request.query_params.get('roles').split(',')
+                queryset = self.filter_users_by_roles(roles, queryset)
+
+            if self.request.query_params.get('languages', None) is not None:
+                if self.request.query_params.get('roles', None) is None or queryset is not None:
+                    languages = self.request.query_params.get('languages').split(',')
+                    queryset = self.filter_users_by_languages(languages, queryset)
+
+            if self.request.query_params.get('sort_by', None) is not None:
+                sort_by = self.request.query_params.get('sort_by').split(',')[
+                    len(self.request.query_params.get('sort_by').split(',')) - 1]
+                sort_by = '-' + sort_by
+
+            else:
+                sort_by = '-id'
+
+            if queryset is None:
+                queryset = User.objects.filter(pk=0)
+
+            return Response(self.serializer_class(queryset.order_by(sort_by), many=True).data)
+
+    def filter_users_by_roles(self, roles, queryset):
+        for role in roles:
+            if self.is_superuser_request(role):
+                queryset = self.get_superusers(queryset)
+            if self.is_staff_request(role):
+                queryset = self.get_staff(queryset)
+            if self.is_training_manager_request(role):
+                queryset = self.get_training_managers(queryset)
+
+        return queryset
+
+    def filter_users_by_languages(self, languages, queryset):
+        if queryset is None:
+            return User.objects.filter(profile__language__in=languages)
+        else:
+            return queryset.filter(profile__language__in=languages)
```

Comment:
can we use DRF backend filter for this? 
https://www.django-rest-framework.org/api-guide/filtering/#djangofilterbackend